tests/test_access_log_parser.py

```diff
@@ -0,0 +1,1132 @@
+    EVENT_SEVERITY,
+    EVENT_SIGNATURE,
+    HTTP_METHOD_DELETE,
+    HTTP_METHOD_GET,
+    HTTP_METHOD_HEAD,
+    HTTP_METHOD_POST,
+    HTTP_METHOD_PUT,
+    HTTP_STATUS_ACCEPTED,
+    HTTP_STATUS_CREATED,
+    HTTP_STATUS_FOUND,
+    HTTP_STATUS_MOVED_PERMANENTLY,
+    HTTP_STATUS_NO_CONTENT,
+    HTTP_STATUS_NOT_FOUND,
+    HTTP_STATUS_OK,
+    HTTP_STATUS_UNAUTHORIZED,
+)
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+FIXTURES_PATH = os.path.join(SCRIPT_DIR, "fixtures")
+
+TEST_REFERRER_ARCHIVEMATICA = "http://test-archivematica-dashboard"
+TEST_REFERRER_STORAGE_SERVICE = "http://test-storage-service"
+
+
+@pytest.mark.parametrize(
+    "fixture_path, expected_return_value",
+    [
+        # Sample log line with authenticated username should be parsed
+        # correctly.
+        (
+            os.path.join(FIXTURES_PATH, "one_line_access_log_with_user.log"),
+            {
+                "ip_address": "172.19.0.1",
+                "remote_user": None,
+                "datetime": "Jan 13 2021 19:53:10",
+                "method": "GET",
+                "url": "/backlog/download/2e28b8a9-351c-4da7-92d2-837ac04cd2d9/",
+                "status_code": "200",
+                "bytes_sent": "19436360",
+                "referrer": "http://127.0.0.1:62080/backlog/",
+                "user_agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0",
+                "forwarded_for": None,
+                "username": "test",
+                "event_name": "Transfer downloaded from backlog",
+                "event_severity": 3,
+                "event_signature": 9,
+                "replaced_uuid": "2e28b8a9-351c-4da7-92d2-837ac04cd2d9",
+                "product": ARCHIVEMATICA_NAME,
+            },
+        ),
+        # Sample log line without authenticated username should also
+        # be parsed correctly, with a username of None.
+        (
+            os.path.join(FIXTURES_PATH, "one_line_access_log_no_user.log"),
+            {
+                "ip_address": "172.19.0.1",
+                "remote_user": None,
+                "datetime": "Jan 13 2021 19:53:10",
+                "method": "GET",
+                "url": "/backlog/download/2e28b8a9-351c-4da7-92d2-837ac04cd2d9/",
+                "status_code": "200",
+                "bytes_sent": "19436360",
+                "referrer": "http://127.0.0.1:62080/backlog/",
+                "user_agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0",
+                "forwarded_for": None,
+                "username": None,
+                "event_name": "Transfer downloaded from backlog",
+                "event_severity": 3,
+                "event_signature": 9,
+                "replaced_uuid": "2e28b8a9-351c-4da7-92d2-837ac04cd2d9",
+                "product": ARCHIVEMATICA_NAME,
+            },
+        ),
+        # Sample log line with URL to filter out should return empty
+        # dictionary.
```

Comment:
It's an interesting question! The way I have [`auditmatica.audit_log._filter_log_line_on_url`](https://github.com/artefactual-labs/auditmatica/blob/386b75c1287746b00e586584ce61b4e2bbd9e122/auditmatica/access_log.py#L182-L206) was inspired by filter functions in the functional programming sense, which is why it returns `True` if the url is one we want to ignore (e.g. is in `URLS_TO_IGNORE` or doesn't have a referrer). But it's not really a filter function per se - it's not taking structure like a list and only returning items that match a certain set of criteria. And we'd typically use a list comprehension for that in Python anyway.

So there might be a better term for it. But I do think what's happening is an active and not passive process - lines are filtered in `auditmatica` not because they don't match the event mappings but because they match specific criteria - which makes me think filtering as a term might still be an okay fit.

Reply:
I've renamed this from `_filter_log_line_on_url` to `_discard_log_line`, which returns a boolean. On further reflection I think you're right and that this is more accurate.